### ISSUE TITLE
RESET track is still created after unchecking "Create RESET track" and adding a second keyframe to the animation

### ISSUE THREAD

**Issue by mcwilleg** [godotengine/godot#56387]
### Godot version

3.4.2.stable

### System information

Windows 10

### Issue description

When a track is created in an AnimationPlayer, and "Create RESET track" is unchecked in the prompt, adding a second key to the same track will create a RESET track anyway.  My expectation is for the RESET track to be ignored for keys on that track permanently, since the prompt is only displayed when the track is created.  If the prompt was displayed every time you added a key to each track, that would be more consistent for the user, however laborious.

### Steps to reproduce

1. Create an AnimationPlayer and any other node to animate.
2. Create a new animation for the node.
3. Click the key button on a property to add a key to the animation, and when the prompt for a new track appears, uncheck "Create RESET track".  
_It doesn't seem to matter what property you animate.  I was able to reproduce this with a basic Transform property track, a Sprite's frame property, as well as custom exported variables from scripts._
4. There should be no RESET track created yet.  Add another key to the same track the same way.
5. The RESET track is created silently (without a prompt).

**Comment by rafallus** [comment]
I've made some research and apparently the problem is this line of code (from master branch)
https://github.com/godotengine/godot/blob/236f4681c9757d3ada0871f090b06250606352be/editor/animation_track_editor.cpp#L3583

There, two conditions are considered to create a RESET track when adding new keys to a track.
1. The track is of a type that can be reset (value, bezier, 3d transform types)
2. The editor parameter 'default_create_reset_tracks' (from Editor>Editor Settings dialog) is set to true.

So, even if the Create RESET track checkbox has been unchecked, the next time a key is added to the track, it creates the RESET track only depending on the value of 'default_create_reset_tracks' parameter.

**Comment by RandomShaper** [comment]
If I've understood the issue correctly, I can say that such behavior is by design.

In cases where there's no confirm dialog, the tool won't assume that the user intent from the last time there was a dialog is relevant. Since you don't have a checkbox this time, the tool considers the most faithful source of truth on what the user wants to do is the `editors/animation/default_create_reset_tracks` setting. Otherwise, to change your mind you'd have to invoke a case where the dialog is shown to be able to change the setting.

That said, that's what I considered when I implemented this, but we can discuss what a better approach could be. I'm not sure that taking last time check box state is superior to the current one.

**Comment by rafallus** [comment]
From the user's point of view, there's a dialog asking if they want to create a RESET track, and if they say no, later a RESET track is created anyway (if `editors/animation/default_create_reset_tracks` is true). That makes the 'Create RESET Track' option kind of pointless, since, at the end, it's reduced to whatever `editors/animation/default_create_reset_tracks` is set to. Would it be better to just rely on the setting value and omit the 'Create RESET Track' checkbox?

**Comment by KoBeWi** [comment]
Yes, make sure to include accurate reproduction steps and some minimal project if relevant.
